modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1311,15 +1313,23 @@ private Predicate _fillPredicate(
 
 			Predicate likePredicate = column.like("%" + search + "%");
 
-			if (predicate == null) {
-				predicate = likePredicate;
+			if (searchPredicate == null) {
+				searchPredicate = likePredicate;
 			}
 			else {
-				predicate = predicate.and(likePredicate);
+				searchPredicate = searchPredicate.or(likePredicate);
 			}
 		}
 
-		return predicate;
+		if (searchPredicate == null) {
+			return predicate;
+		}
+
+		if (predicate == null) {
+			return searchPredicate;
```

Comment:
Hi @MarinhoFeliphe, 
Don't you think that returning this variable is the same as return null? Why did you prefer to return this var instead of returning null?
`if (predicate == null) {
	return searchPredicate;
}`